godot-core/src/obj/gd.rs

```diff
@@ -444,6 +445,21 @@ impl<T: GodotClass> Gd<T> {
     pub fn callable<S: Into<StringName>>(&self, method_name: S) -> Callable {
         Callable::from_object_method(self.share(), method_name)
     }
+
+    /// Returns whether `T` inherits from `U`.
+    ///
+    /// This is reflexive, i.e `T` inherits from itself.
+    ///
+    /// See also [`Inherits`] for a trait bound.
+    pub fn inherits<U: GodotClass>() -> bool {
+        if T::CLASS_NAME == U::CLASS_NAME {
+            true
+        } else if T::Base::CLASS_NAME == <()>::CLASS_NAME {
+            false
+        } else {
+            Gd::<T::Base>::inherits::<U>()
+        }
+    }
```

Comment:
Quite a clever recursive implementation! 💡 

If we want to provide this, it should probably being inherent on `T` and not `Gd<T>`. The usage below would also support this. There are 3 options:
1. global function `inherits::<T, Base>()`
2. `GodotClass` default impl `T::inherits::<Base>()`
3. extension trait `T::inherits::<Base>()` -- although I don't like defining a new trait just for this

Reply:
makes sense, I'd probably go for the second option. it makes sense to me that a trait for classes would have such a function.